modules/features/endofyear/src/main/java/au/com/shiftyjelly/pocketcasts/endofyear/StoriesFragment.kt

```diff
@@ -108,12 +140,33 @@ class StoriesFragment : BaseAppCompatDialogFragment() {
         super.onDismiss(dialog)
     }
 
+    private fun startUpsellFlow() {
+        val flow = OnboardingFlow.Upsell(OnboardingUpgradeSource.END_OF_YEAR)
+        OnboardingLauncher.openOnboardingFlow(requireActivity(), flow)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        viewModel.resumeStoryAutoProgress()
+    }
+
+    override fun onPause() {
+        // Pause auto progress when the fragment is not active.
+        // This makes sure that users see all stories and they
+        // won't auto switch for example when signing up takes
+        // some time or when the EoY flow is interruped by
+        // some other user actions such as a phone call.
+        viewModel.pauseStoryAutoProgress()
```

Comment:
This might already be coming in your next PR: We should also allow pausing a story by long-pressing it to enable a user to stay on the current story for a longer time.

Reply:
Thanks! I didn't know. It wasn't specified anywhere. I'll confirm with David and add it to the requirements.

Reply:
Just to clarify. Do you mean that holding the story should pause it? And not that long pressing turns on and off the auto progress?

Reply:
> Do you mean that holding the story should pause the progress? 

That's correct, and progress should resume when you un-touch the story.
We did that in previous years. Not sure if it is needed now.